File: backend/app/schemas/ai_video.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class AiVideoJobCreate(BaseModel):
    script: str = Field(..., min_length=2)
    title: Optional[str] = None
    videoType: str = "knowledge_visualization"
    style: str = "futuristic"
    aspectRatio: str = "16:9"
    voiceProvider: str = "cosyvoice"
    voiceId: str = "中文女"
    subtitleMode: str = "keywords"
    brandKitId: Optional[str] = None
    targetPlatform: str = "douyin"


class AiVideoJobCreated(BaseModel):
    jobId: str
    projectId: int
    status: str


class AiVideoJobResponse(BaseModel):
    jobId: str
    id: int
    projectId: int
    status: str
    progress: int
    stage: str
    message: str
    outputUrl: Optional[str]
    coverUrl: Optional[str]
    errorMessage: Optional[str]
    createdAt: datetime
    updatedAt: datetime
    completedAt: Optional[datetime]


class AiVideoProjectResponse(BaseModel):
    id: int
    title: str
    videoType: str
    aspectRatio: str
    status: str
    coverUrl: Optional[str]
    outputUrl: Optional[str]
    currentVersionId: Optional[int]
    createdAt: datetime
    updatedAt: datetime
    projectJson: Optional[dict[str, Any]] = None


class AiVideoEditRequest(BaseModel):
    message: str = Field(..., min_length=2)
    mode: str = "plan_then_apply"


class AiVideoEditPlanResponse(BaseModel):
    editPlan: list[str]
    canApply: bool = True


class AiVideoApplyEditRequest(BaseModel):
    editPlan: list[str] = Field(default_factory=list)
    message: Optional[str] = None


class AiVideoBrandKitCreate(BaseModel):
    name: str
    colors: list[str] = Field(default_factory=lambda: ["#0f766e", "#2563eb", "#f59e0b"])
    voiceConfig: dict[str, Any] = Field(default_factory=dict)
    subtitleStyle: dict[str, Any] = Field(default_factory=dict)


class AiVideoBrandKitResponse(BaseModel):
    id: int
    name: str
    logoUrl: Optional[str]
    colors: list[str]
    voiceConfig: dict[str, Any]
    subtitleStyle: dict[str, Any]
    createdAt: datetime
    updatedAt: datetime

File: frontend/src/services/aiVideo.ts
import api from './api'

export interface AiVideoJobCreate {
  title?: string
  prompt?: string
  script: string
  videoType: string
  contentType?: string
  style: string
  visualStyle?: string
  aspectRatio: string
  voiceProvider: string
  voiceId: string
  subtitleMode: string
  brandKitId?: string
  targetPlatform: string
  tone?: string
  pace?: string
  goal?: string
  customPrompt?: string
  sceneCount?: number
  draftScenes?: any[]
}

export interface AiVideoJob {
  jobId: string
  id: number
  projectId: number
  status: string
  progress: number
  stage: string
  message: string
  outputUrl: string | null
  coverUrl: string | null
  errorMessage: string | null
  createdAt: string
  updatedAt: string
  completedAt: string | null
}

export interface AiVideoProject {
  id: number
  title: string
  videoType: string
  aspectRatio: string
  status: string
  coverUrl: string | null
  outputUrl: string | null
  currentVersionId: number | null
  createdAt: string
  updatedAt: string
  projectJson?: any
}

export interface AiVideoOverview {
  stats: {
    totalProjects: number
    generating: number
    completed: number
    monthlyExports: number
  }
  recentProjects: AiVideoProject[]
  queue: AiVideoJob[]
}

export interface AiVideoCapability {
  module: string
  storageRoot: string
  renderServiceUrl: string
  renderService: { available: boolean; status?: any; message?: string }
  cosyVoiceService?: { available: boolean; status?: any; message?: string }
  ffmpeg: { available: boolean; path?: string | null }
  isolation: Record<string, any>
  limits: Record<string, any>
}

export interface AiVideoExport {
  id: number
  projectId: number
  title: string
  versionNo: number
  outputUrl: string | null
  coverUrl: string | null
  status: string
  createdAt: string
}

export const aiVideoApi = {
  overview: () => api.get<AiVideoOverview>('/ai-video/overview'),
  capabilities: () => api.get<AiVideoCapability>('/ai-video/capabilities'),
  templates: () => api.get('/ai-video/templates'),
  storyboardDraft: (payload: AiVideoJobCreate) => api.post<{ projectJson: any; scenes: any[]; recommendations: string[] }>('/ai-video/storyboard-draft', payload),
  createJob: (payload: AiVideoJobCreate) => api.post<{ jobId: string; projectId: number; status: string }>('/ai-video/jobs', payload),
  getJob: (jobId: string) => api.get<AiVideoJob>(`/ai-video/jobs/${jobId}`),
  cancelJob: (jobId: string) => api.post<AiVideoJob>(`/ai-video/jobs/${jobId}/cancel`),
  retryJob: (jobId: string) => api.post<{ jobId: string; projectId: number; status: string }>(`/ai-video/jobs/${jobId}/retry`),
  listProjects: () => api.get<AiVideoProject[]>('/ai-video/projects'),
  getProject: (id: number) => api.get<AiVideoProject>(`/ai-video/projects/${id}`),
  planEdit: (id: number, message: string) => api.post<{ editPlan: string[]; canApply: boolean }>(`/ai-video/projects/${id}/edit`, { message, mode: 'plan_then_apply' }),
  applyEdit: (id: number, editPlan: string[], message?: string) => api.post(`/ai-video/projects/${id}/apply-edit`, { editPlan, message }),
  versions: (id: number) => api.get(`/ai-video/projects/${id}/versions`),
  rollbackVersion: (projectId: number, versionId: number) => api.post(`/ai-video/projects/${projectId}/versions/${versionId}/rollback`),
  exports: () => api.get<AiVideoExport[]>('/ai-video/exports'),
  brandKits: () => api.get('/ai-video/brand-kits'),
  createBrandKit: (payload: { name: string; colors: string[] }) => api.post('/ai-video/brand-kits', payload),
}
